src/Stratis.Bitcoin.Features.BlockStore/BlockStoreBehavior.cs

```diff
@@ -180,13 +182,10 @@ private ChainedHeader GetBestHeader()
         {
             this.logger.LogTrace("()");
 
-            //TODO use CMB.BestPeerTip instead. Also when we update LastHeaderSent update CMB.UpdateBestHeaderSent
-            ChainedHeader peerTip = this.consensusManagerBehavior.ExpectedPeerTip;
+            ChainedHeader bestHeader = this.consensusManagerBehavior.BestSentHeader;
 
-            ChainedHeader bestHeader = peerTip;
-
-            if ((bestHeader == null) || ((this.lastHeaderSent != null) && (bestHeader.Height < this.lastHeaderSent.Height)))
-                bestHeader = this.lastHeaderSent;
+            if ((bestHeader == null) || ((this.lastSentHeader != null) && (bestHeader.Height < this.lastSentHeader.Height)))
```

Comment:
Maybe we should compare chain work instead

Reply:
Actually we should not need to have this at all, these two lines should be removed